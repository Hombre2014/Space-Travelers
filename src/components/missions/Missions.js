import { React, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { v4 as uuidv4 } from 'uuid';
import Mission from './Mission';
import missionsFetch from '../../redux/missions/missionsFetch';
import './missions.css';

const Missions = () => {
  const missions = useSelector((state) => state.missionsReducer);
  const dispatch = useDispatch();

  useEffect(() => { dispatch(missionsFetch()); }, []);

  return (
    <div className="container">
      <table className="table">
        <tbody>
          <tr>
            <th className="head">Missions</th>
            <th className="head">Description</th>
            <th className="head">Status</th>
            <th className="head">Action</th>
          </tr>
          {missions.map(
            (mission) => (
              <Mission
                key={uuidv4()}
                mission_name={mission.mission_name}
                description={mission.description}
              />
            ),
          )}
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
