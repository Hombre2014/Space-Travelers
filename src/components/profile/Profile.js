import React from 'react';
import { useSelector } from 'react-redux';
import { v4 as uuidv4 } from 'uuid';
import './profile.css';

const Profile = () => {
  const availabeRockets = useSelector((state) => state.rocketsReducer);
  const missions = useSelector((state) => state.missionsReducer);
  return (
    <div className="container">
      <section className="missions">
        <h2>My Missions</h2>
        <table className="missions-table">
          <tbody>
            {missions
              && missions
                .filter((mission) => mission.reserved === true)
                .map((reserve) => (
                  <tr className="row" key={reserve.mission_id}>
                    <td>{reserve.mission_name}</td>
                  </tr>
                ))}
          </tbody>
        </table>
      </section>
      <section>
        <h2>My Rockets</h2>
        <ul>
          {availabeRockets
            && availabeRockets
              .filter((rocket) => rocket.reserved === true)
              .map((reserved) => (
                <li key={uuidv4()}>
                  {reserved.name}
                </li>
              ))}
        </ul>
      </section>
    </div>
  );
};

export default Profile;
