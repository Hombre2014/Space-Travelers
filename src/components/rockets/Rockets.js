import React from 'react';
import { useSelector } from 'react-redux';
import { v4 as uuidv4 } from 'uuid';
import Rocket from './Rocket';

function Rockets() {
  const availableRockets = useSelector((state) => state.rocketsReducer);

  return (
    <div>
      { availableRockets.map((rocket) => (
        <Rocket
          key={uuidv4()}
          id={rocket.id}
          name={rocket.name}
          description={rocket.description}
          images={rocket.images[0]}
          reserved={rocket.reserved}
        />
      ))}
    </div>
  );
}

export default Rockets;
