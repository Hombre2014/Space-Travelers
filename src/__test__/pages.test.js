import React from 'react';
import { render } from '@testing-library/react';
import Rockets from '../components/rockets/Rockets';
import Missions from '../components/missions/Missions';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Profile from '../components/profile/Profile';

describe('Testing if all pages renders correctly', () => {
  test('Testing Rockets page', () => {
    const tree = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('Testing Missions page', () => {
    const tree = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('Testing Profile page', () => {
    const tree = render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
