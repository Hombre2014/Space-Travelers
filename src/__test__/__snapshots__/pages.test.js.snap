// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if all pages renders correctly Testing Missions page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <table
          class="table"
        >
          <tbody>
            <tr>
              <th
                class="head"
              >
                Missions
              </th>
              <th
                class="head"
              >
                Description
              </th>
              <th
                class="head"
              >
                Status
              </th>
              <th
                class="head"
              >
                Action
              </th>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <table
        class="table"
      >
        <tbody>
          <tr>
            <th
              class="head"
            >
              Missions
            </th>
            <th
              class="head"
            >
              Description
            </th>
            <th
              class="head"
            >
              Status
            </th>
            <th
              class="head"
            >
              Action
            </th>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing if all pages renders correctly Testing Profile page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <section
          class="missions"
        >
          <h2>
            My Missions
          </h2>
          <table
            class="missions-table"
          >
            <tbody />
          </table>
        </section>
        <section
          class="missions"
        >
          <h2>
            My Rockets
          </h2>
          <table
            class="rockets-table"
          >
            <tbody />
          </table>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <section
        class="missions"
      >
        <h2>
          My Missions
        </h2>
        <table
          class="missions-table"
        >
          <tbody />
        </table>
      </section>
      <section
        class="missions"
      >
        <h2>
          My Rockets
        </h2>
        <table
          class="rockets-table"
        >
          <tbody />
        </table>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing if all pages renders correctly Testing Rockets page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div />
    </div>
  </body>,
  "container": <div>
    <div />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
